components/datetime/src/time_zone.rs

```diff
@@ -1257,7 +1234,7 @@ impl FormatTimeZone for Iso8601Format {
                         return Ok(Err(e));
                     }
                 }
-                if let Err(e) = sink.write_str(&self.format_offset_seconds(time_zone)) {
+                if let Err(e) = sink.write_str(&TimeZoneFormat::format_offset_seconds(time_zone)) {
```

Comment:
Issue: `format_offset_seconds` allocates memory. You should avoid the allocation cost. To do this, make `format_offset_seconds` take a sink, and then do `TimeZomeFormat::format_offset_seconds(time_zone, sink)`

Reply:
hmm, do you mean that the returned `String` requires more allocation cost? for example, it gets assigned to a temporary variable and then the temporary variable is passed to `sink.write_str`?